stl/inc/future

```diff
@@ -1416,6 +1416,18 @@ private:
     _MyPromiseType _MyPromise;
 };
 
+#if _HAS_CXX17

```

Comment:
I this really C++17 or should this be for C++20?

Reply:
It would be good to see maintainer's opinion. 
I guess it's appropriate for C++17, but I'm not good in standardization procedures, so can't say anything factual

Reply:
Our general rule is to implement LWG issue resolutions as unconditionally as possible, unless we discover (or clearly anticipate) that they are obnoxiously source-breaking (which is very rare). In this case, adding deduction guides is highly non-intrusive - they have no effect until a user intentionally uses CTAD, in which case these guides are needed to get a sensible result. Because CTAD is a C++17 feature, guarding this with `_HAS_CXX17` is exactly correct. This also aligns with the practice of other implementers and the sense of the LWG that all issue resolutions are defect reports (even if that may not formally be the case).

(Note that this is the opposite of how we implement features, where we guard them with the corresponding Standard version unless there is a clear reason to implement them unconditionally.)